File: frontend/src/pages/Sentiment.js
import React, { useState } from 'react';
import axios from 'axios';
import './Sentiment.css';

const GATEWAY_URL = process.env.REACT_APP_GATEWAY_URL || 'http://localhost:8000';

function Sentiment() {
  const [text, setText] = useState('');
  const [result, setResult] = useState(null);
  const [loading, setLoading] = useState(false);
  const [error, setError] = useState(null);

  const analyzeSentiment = async () => {
    if (!text.trim()) {
      setError('Please enter some text to analyze');
      return;
    }

    setLoading(true);
    setError(null);
    setResult(null);

    try {
      const response = await axios.post(`${GATEWAY_URL}/sentiment`, {
        text: text
      });

      setResult(response.data);
    } catch (err) {
      setError(err.response?.data?.detail || 'Failed to analyze sentiment. Please try again.');
    } finally {
      setLoading(false);
    }
  };

  return (
    <div className="sentiment-page">
      <div className="page-header">
        <h1>😊 Sentiment Analysis</h1>
        <p>Analyze the emotional tone of any text using AI-powered sentiment detection</p>
      </div>

      <div className="input-section">
        <h3>Enter Your Text</h3>
        <textarea
          className="text-input"
          rows="6"
          placeholder="Type or paste any text here... (e.g., 'I absolutely loved this product! It exceeded all my expectations.')"
          value={text}
          onChange={(e) => setText(e.target.value)}
        />
        <button 
          className="analyze-btn" 
          onClick={analyzeSentiment}
          disabled={loading}
        >
          {loading ? '🔄 Analyzing...' : '🔍 Analyze Sentiment'}
        </button>
      </div>

      {loading && (
        <div className="loading">
          <p>Analyzing sentiment...</p>
        </div>
      )}

      {error && (
        <div className="error">
          <strong>Error:</strong> {error}
        </div>
      )}

      {result && (
        <div className="result-section">
          <div className={result.label === 'POSITIVE' ? 'result-positive' : 'result-negative'}>
            <div className="result-emoji">
              {result.label === 'POSITIVE' ? '😊' : '😞'}
            </div>
            <div className="result-label">{result.label} Sentiment</div>
            <div className="result-score">{(result.score * 100).toFixed(1)}% Confidence</div>
          </div>

          <div className="progress-bar">
            <div 
              className="progress-fill" 
              style={{width: `${result.score * 100}%`}}
            >
              {(result.score * 100).toFixed(2)}%
            </div>
          </div>

          <div className="analyzed-text">
            <h4>📝 Analyzed Text</h4>
            <p>{text}</p>
          </div>
        </div>
      )}
    </div>
  );
}

export default Sentiment;

File: streamlit_app/pages/sentiment.py
import streamlit as st
import requests
import os
from dotenv import load_dotenv

# Load environment variables
load_dotenv()

# -----------------------------
# Configuration
# -----------------------------
GATEWAY_URL = os.getenv("GATEWAY_URL", "http://localhost:8000")

st.set_page_config(
    page_title="Sentiment Analysis",
    page_icon="😊",
    layout="wide" 
)

# Custom CSS
st.markdown("""
<style>
    .sentiment-header {
        text-align: center;
        padding: 1.5rem;
        background: linear-gradient(135deg, #CD0000 0%, #FF4444 100%);
        color: #EFEDE6;
        border-radius: 10px;
        margin-bottom: 2rem;
    }
    .result-card {
        padding: 2rem;
        border-radius: 10px;
        background: #EFEDE6;
        margin: 1rem 0;
    }
    .metric-positive {
        background: linear-gradient(135deg, #11998e 0%, #38ef7d 100%);
        padding: 2rem;
        border-radius: 10px;
        color: #EFEDE6;
        text-align: center;
    }
    .metric-negative {
        background: linear-gradient(135deg, #CD0000 0%, #FF4444 100%);
        padding: 2rem;
        border-radius: 10px;
        color: #EFEDE6;
        text-align: center;
    }
    .stButton>button {
        background-color: #CD0000;
        color: #EFEDE6;
    }
    .stButton>button:hover {
        background-color: #FF4444;
    }
</style>
""", unsafe_allow_html=True)

st.markdown("""
<div class="sentiment-header">
    <h1>😊 Sentiment Analysis</h1>
    <p>Analyze the emotional tone of any text using AI-powered sentiment detection</p>
</div>
""", unsafe_allow_html=True)

st.markdown("### Enter Your Text")
st.markdown("Type or paste any text below to analyze its sentiment (positive or negative).")

# -----------------------------
# Input
# -----------------------------
text = st.text_area(
    "",
    height=180,
    placeholder="Example: I absolutely loved this product! It exceeded all my expectations.",
    label_visibility="collapsed"
)

st.markdown("")
col1, col2, col3 = st.columns([2, 1, 2])

with col2:
    analyze = st.button(
        "🔍 Analyze Sentiment",
        type="primary"
    )

# -----------------------------
# Prediction
# -----------------------------
if analyze:

    if not text.strip():
        st.warning("Please enter some text.")
        st.stop()

    payload = {
        "text": text
    }

    try:

        with st.spinner("Analyzing sentiment..."):

            response = requests.post(
                f"{GATEWAY_URL}/sentiment",
                json=payload,
                timeout=30
            )

        if response.status_code == 200:

            data = response.json()

            label = data.get("label", "Unknown")
            confidence = data.get("score", 0)

            st.success("✅ Analysis Complete!")

            st.markdown("")

            # Display results based on sentiment
            if label.upper() == "POSITIVE":
                st.markdown("""
                <div class="metric-positive">
                    <h1 style="font-size: 4rem; margin: 0;">😊</h1>
                    <h2>POSITIVE Sentiment</h2>
                    <p style="font-size: 2rem; margin-top: 1rem; font-weight: bold;">{:.1f}% Confidence</p>
                </div>
                """.format(confidence * 100), unsafe_allow_html=True)
            else:
                st.markdown("""
                <div class="metric-negative">
                    <h1 style="font-size: 4rem; margin: 0;">😞</h1>
                    <h2>NEGATIVE Sentiment</h2>
                    <p style="font-size: 2rem; margin-top: 1rem; font-weight: bold;">{:.1f}% Confidence</p>
                </div>
                """.format(confidence * 100), unsafe_allow_html=True)

            st.markdown("")
            
            # Progress bar
            st.progress(float(confidence), text=f"Model Confidence: {confidence*100:.2f}%")

            st.markdown("")
            st.markdown("### 📝 Analyzed Text")
            st.info(text)

        else:

            st.error(
                f"Gateway returned {response.status_code}"
            )

            st.code(response.text)

    except requests.exceptions.ConnectionError:

        st.error(
            """
Unable to connect to the Gateway.

Make sure your FastAPI Gateway is running.

Expected URL:

http://localhost:8000
"""
        )

    except Exception as e:

        st.exception(e)